code-formatting/pre-commit.sh

```
@@ -5,10 +5,72 @@ format_cmd=""
+        exit 1
+    fi
+else
+    echo '⚠ gitleaks not installed — skipping secret scan. Install: brew install gitleaks' 1>&2
+fi
+
+# --- Block binary blobs and oversized files -------------------------------
+STAGED=$(git diff --staged --name-only --diff-filter=ACMR)
+JAVABIN=$(printf '%s\n' "$STAGED" | grep -i '\.javabin$' || true)
+if [ -n "$JAVABIN" ]; then
+    echo '✖ Refusing to commit .javabin data file(s):' 1>&2
+    printf '    %s\n' $JAVABIN 1>&2
```

Shellcheck (0.11.0)

[info] Double quote to prevent globbing and word splitting.

(SC2086)